server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/mailet/filter/ActionApplier.java

```diff
@@ -101,6 +101,17 @@ private void addStorageDirective(Rule.Action action) {
                 .collect(ImmutableList.toImmutableList()));
             return;
         }
+        if (action.getForward().isPresent()) {
+            Rule.Action.Forward forward = action.getForward().get();
+            if (!forward.isKeepACopy()) {
+                mail.setRecipients(mail.getRecipients().stream()
+                    .filter(recipient -> !recipient.equals(mailAddress))
+                    .collect(ImmutableList.toImmutableList()));
+            }
+            mail.setRecipients(Stream.concat(mail.getRecipients().stream(), forward.getAddresses().stream())
+                .collect(ImmutableList.toImmutableList()));
```

Comment:
No!

We need to use the mailetContext to send a copy of this email to the sole forwards. We also need to change the sender to the current recipient to not have DKIM / SPF problems.

CF JAMES-3944 and https://github.com/apache/james-project/pull/1750